src/lib.rs

```diff
@@ -346,10 +345,425 @@ impl From<Error> for io::Error {
     }
 }
 
+/// Methods that return information about a path.
+///
+/// This trait provides the familiar methods from `std::path::Path`
+/// for the `Path*` types. These methods take the same parameters and return
+/// the same types as the originals in the standard library, except where
+/// noted.
+///
+/// As a general rule, methods that can return an error will return a rich
+/// [`path_abs::Error`] instead of a [`std::io::Error`].
+///
+/// [`path_abs::Error`]: struct.Error.html
+/// [`std::io::Error`]: https://doc.rust-lang.org/stable/std/io/struct.Error.html
+pub trait PathInfo: Clone + Borrow<PathBuf> + Into<Arc<PathBuf>> {
```

Comment:
is the `Into` necessary here? Specifically, does `Path` satisfy this?

I would like to implement these types for `Path` (obviously not `PathMut` though).

Reply:
I originally tried to design these traits so that they could work for both `&Path` and `PathBuf`, but very quickly gave up, because there's a fundamental difference between what you can do with an owned type (even one containing a reference like `Foo<'a>`) and a borrowed one.

That said, I have changed the trait design a couple of times since I gave up on supporting `Path`, maybe it'd work better now.

Reply:
Turns out, yeah, with a few small changes it's possible to implement `PathInfo` for `&Path`. See bb40d46.